assets/js/blocks/product-search/block.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { Component } from '@wordpress/element';
+import { IconButton } from '@wordpress/components';
+import PropTypes from 'prop-types';
+import { withInstanceId, compose } from '@wordpress/compose';
+import { RichText } from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import './style.scss';
+
+/**
+ * Component displaying a product search form.
+ */
+class ProductSearchBlock extends Component {
+	render() {
+		if ( this.props.isPreview ) {
+			return this.renderEdit();
+		}
+
+		return this.renderView();
+	}
+
+	renderView() {
+		const { attributes: { label, placeholder, formId, className, hasLabel } } = this.props;
+		const home = wc_product_block_data.homeUrl;
+		const classes = classnames(
+			'wc-block-product-search',
+			className,
+		);
+		const data = {};
+
+		if ( hasLabel ) {
+			data[ 'data-has-label' ] = true;
+		}
+
+		return (
+			<form className={ classes } { ...data } role="search" method="get" action={ home } data-form-id={ formId }>
+				<label
+					htmlFor={ formId }
+					className={ !! hasLabel ? 'wc-block-product-search__label' : 'wc-block-product-search__label screen-reader-text' }
+				>
+					{ label }
+				</label>
+				<div className="wc-block-product-search__fields">
+					<input
+						type="search"
+						id={ formId }
+						className="wc-block-product-search__field"
+						placeholder={ placeholder }
+						name="s"
+					/>
+					<input type="hidden" name="post_type" value="product" />
+					<IconButton
+						icon="arrow-right-alt2"
+						className="wc-block-product-search__button"
+						label={ __( 'Search', 'woo-gutenberg-products-block' ) }
+						type="submit"
+					/>
+				</div>
+			</form>
+		);
+	}
+
+	renderEdit() {
+		const { attributes, setAttributes, instanceId } = this.props;
+		const { label, placeholder, formId, className, hasLabel } = attributes;
+		const classes = classnames(
+			'wc-block-product-search',
+			className,
+		);
+
+		if ( ! formId ) {
+			setAttributes( { formId: `wc-block-product-search-${ instanceId }` } );
+		}
+
+		return (
+			<form className={ classes }>
+				{ !! hasLabel && (
+					<RichText
```

Comment:
When clicking on this RichText it selects all the text and I can't leave the pointer to a specific position, I have to use the keyboard to do that. It's really weird, does the same happen to you?

![Peek 2019-07-08 17-28](https://user-images.githubusercontent.com/3616980/60824946-dfa2a000-a1aa-11e9-9806-9883195c3af2.gif)

Reply:
Had to switch to a span but this is resolved.